spec/classes/init_spec.rb

```diff
@@ -279,6 +280,28 @@
           }
         end
 
+        context 'when enabling nspawn' do
+          let(:params) do
+            {
+              manage_nspawn: true,
+            }
+          end
+
+          case facts[:os]['family']
+          when 'RedHat'
+            case facts[:os]['release']['major']
+            when '7'
+              it { is_expected.not_to contain_package('systemd-container') } # rubocop:disable RSpec/RepeatedExample
+            else
+              it { is_expected.to contain_package('systemd-container').with_ensure('present') } # rubocop:disable RSpec/RepeatedExample
+            end
+          when 'Debian'
+            it { is_expected.to contain_package('systemd-container').with_ensure('present') } # rubocop:disable RSpec/RepeatedExample
+          else
+            it { is_expected.not_to contain_package('systemd-container') } # rubocop:disable RSpec/RepeatedExample
+          end
```

Comment:
Is this better?
```suggestion
          # rubocop:disable RSpec/RepeatedExample
          case facts[:os]['family']
          when 'RedHat'
            case facts[:os]['release']['major']
            when '7'
              it { is_expected.not_to contain_package('systemd-container') }
            else
              it { is_expected.to contain_package('systemd-container').with_ensure('present') }
            end
          when 'Debian'
            it { is_expected.to contain_package('systemd-container').with_ensure('present') }
          else
            it { is_expected.not_to contain_package('systemd-container') }
          end
          # rubocop:enable RSpec/RepeatedExample
```

Reply:
I'd say the precise line comments was clearer as to where the "offence" is, more obvious what to do when EL7 dies.